src/dir.targets

```diff
@@ -187,6 +203,36 @@
     <Error Condition="'$(TestRunExitCode)' != '0'" Text="One or more tests failed while running tests from '$(MSBuildProjectName)' please check log for details!" />
   </Target>
 
+  <Target Name="CheckTestCategories"
+          AfterTargets="Test"
+          Condition="'$(TestDisabled)'!='True'">
+
+    <Error Condition="'$(RunTestsWithCategories)' != '' And '$(RunTestsWithoutCategories)' != ''"
+           Text="Specifying both RunTestsWithCategories and RunTestsWithoutCategories is ambiguous." />
+
+    <ItemGroup Condition="'@(RunTestsWithCategoriesItems)'!=''">
+      <!-- Intersection of TestCategoriesItems and RunTestsWithCategoriesItems -->
+      <TestCategoriesToRun Include="@(TestCategoriesItems)" Condition="'@(RunTestsWithCategoriesItems)'=='%(Identity)'" />
+    </ItemGroup>
+    <ItemGroup Condition="'@(RunTestsWithoutCategoriesItems)'!=''">
```

Comment:
I would combine both of these item groups given they have the same condition. It is good to eliminate the duplication.